MREUnityRuntime/MREUnityRuntimeLib/Util/Logging/UnityLogger.cs

```diff
@@ -1,28 +1,46 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+using MixedRealityExtension.App;
 using MixedRealityExtension.PluginInterfaces;
+using MixedRealityExtension.Messaging.Payloads;
+using MixedRealityExtension.Messaging;
 using UnityEngine;
-
 namespace MixedRealityExtension.Util.Logging
 {
     internal class UnityLogger : IMRELogger
     {
+        MixedRealityExtensionApp _app;
+        public UnityLogger(MixedRealityExtensionApp app)
+        {
+            _app = app;
+        }
+        void Send(TraceSeverity severity, string message)
+        {
+            Traces traces = new Traces();
```

Comment:
This is a lot of allocations here.  Every single log will now result in an allocation of the payload, the internal list of traces, and the single trace to send.  You should make a private member field that is the Traces payload and just reuse the one trace that will be in the list inside that payload.  Far more efficient.